src/main/java/com/datastax/cdm/cql/statement/TargetUpdateStatement.java

```diff
@@ -42,35 +42,39 @@ protected BoundStatement bind(Row originRow, Row targetRow, Integer ttl, Long wr
             boundStatement = boundStatement.set(currentBindIndex++, writeTime, Long.class);
         }
 
-        Object bindValue = null;
+        Object bindValueOrigin = null;
+        Object bindValueTarget = null;
         for (int targetIndex : columnIndexesToBind) {
             int originIndex = cqlTable.getCorrespondingIndex(targetIndex);
 
             try {
-                if(usingCounter && counterIndexes.contains(targetIndex)) {
-                    bindValue = cqlTable.getOtherCqlTable().getData(originIndex, originRow);
-                    if (null == bindValue) {
+                if (usingCounter && counterIndexes.contains(targetIndex)) {
+                    bindValueOrigin = cqlTable.getOtherCqlTable().getData(originIndex, originRow);
+                    if (null == bindValueOrigin) {
                         currentBindIndex++;
                         continue;
                     }
-                    bindValue = ((Long) bindValue - (null==targetRow ? 0 : (Long) cqlTable.getData(targetIndex, targetRow)));
+                    bindValueTarget = (null == targetRow ? 0L : cqlTable.getData(targetIndex, targetRow));
+                    bindValueTarget = ((Long) bindValueOrigin - (null == bindValueTarget ? 0L : (Long) bindValueTarget));
```

Comment:
I don't understand this fully - we've done null protection so why not just reset to bindValueOrigin - bindValueTarget?  And what if origin value is null? Discussion may clear this up...

Reply:
Ah I think I see it now, some code comments would be good...
First one is "what is the value on target, if the whole targetRow does not exist use a value of 0". Second one is "compute if the value itself on a non-null targetRow is null".

Reply:
Agree this is a bit confusing, but most of the confusion is due to how counter tables work in Cassandra & how nulls work in Java. I find looking at actual code to be the best docs in such scenarios.

Reply:
Yes but in this case the code confused me as we set `bindValueTarget` twice :)  First entry could be maybe `long bindValueTargetFromRow`  if we wanted to use code as documentation?

Reply:
The first is handling if the entire row is null & the second is handling if the column value is null. But in both scenarios the variable is storing the target column values (not row). I have updated the variable naming for the whole section to be more clear.